api/src/main/java/org/neo4j/ogm/model/Edge.java

```diff
@@ -35,4 +35,15 @@
      */
     String getPrimaryIdName();
 
+    /**
+     * Returns if the relationship entity has version property
+     */
+    boolean hasVersionProperty();
```

Comment:
It is present both on Edge and Node because each node/edge (edge based on relationship entity only) has its own version.

Maybe we could extract _some_ of this to new metadata class and share it between individual NodeModel/EdgeModel instances.